src/main/scala/com/evolution/scache/ExpiringCache.scala

```diff
@@ -40,16 +41,17 @@ object ExpiringCache {
       def removeExpired(key: K, entryRef: LoadingCache.EntryRef[F, Entry[V]]) = {
         entryRef
           .get
-          .flatMap { entry =>
-            entry.foldMapM { entry =>
+          .flatMap {
+            case EntryState.Value(entry) =>
               for {
                 now               <- Clock[F].millis
                 expiredAfterRead   = expireAfterReadMs + entry.value.touched < now
                 expiredAfterWrite  = () => expireAfterWriteMs.exists { _ + entry.value.created < now }
                 expired            = expiredAfterRead || expiredAfterWrite()
                 result            <- if (expired) remove(key) else ().pure[F]
               } yield result
-            }
+            case EntryState.Loading(_) => ().pure[F]
```

Comment:
please use `_: EntryState.Loading / _: EntryState.Value / _: EntryState.Removed`, so we do not need to adjust 100500 places after adding new fields

Reply:
It would have to be `_: EntryState.Loading[F, V]`, etc. Do you still think it would be better?